Plutarch/Rec/TH.hs

```diff
@@ -15,10 +16,13 @@ deriveAll name = (<>) <$> deriveScottEncoded name <*> Rank2.TH.deriveAll name
 -- | Use as a TH splice for @type instance ScottEncoded@ declarations.
 deriveScottEncoded :: TH.Name -> Q [TH.Dec]
 deriveScottEncoded name = do
-  con <- reifyConstructor name
+  (con, tyVars) <- reifyConstructor name
```

Comment:
?

Reply:
There was a bug in TH generation for types with parameters additional to `(s :: S)`, such as the `PValue (keys :: ValueKeyGuarantees) (amounts :: ValueAmountGuarantees) (s :: S)`.

Reply:
This should be a separate PR really

Reply:
It was, PR #453.